### PR TITLE
MOSIP-11446: Storing Operator biometric templates in DB

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**

### CODE REVIEW

File: registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java [hunk 1]
```diff
@@ -368,7 +369,19 @@ private ResponseDTO save(List<BiometricsDto> biometrics) {
...
-			onBoardingResponse = userOnBoardDao.insert(biometrics);
+			List<BIR> birList = new ArrayList<>();
+			for (BiometricsDto biometricsDto : biometrics) {
+				BIR bir = buildBir(biometricsDto);
+				LOGGER.debug(LOG_REG_USER_ONBOARD, APPLICATION_NAME, APPLICATION_ID, "Adding bir");
+				birList.add(bir);
+			}
+
+			List<BIR> templates = bioAPIFactory
+					.getBioProvider(BiometricType.fromValue(birList.get(0).getBdbInfo().getType().get(0).value()),
```
Comment (ghost) [inline registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java]: You are taking BiometricType of first element and passing biometrics of other biometricType as well to extract method. We should get bioPriovder for each BiometricType separately and then invoke extract with biometrics related to corresponding BiometricType.

**Review by ghost — CHANGES_REQUESTED** [review]

File: registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java (hunk 1 above)
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java]: Modified to get bioProvider for each BiometricType separately

File: registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java [hunk 2]
```diff
@@ -106,18 +108,16 @@
...
-				"validateWithIDAuthAndSave invoked ");
+		LOGGER.info(LOG_REG_USER_ONBOARD, APPLICATION_NAME, APPLICATION_ID, "validateWithIDAuthAndSave invoked ");
 
 		if (Objects.isNull(biometrics))
 			throw new RegBaseCheckedException(RegistrationExceptionConstants.REG_BIOMETRIC_DTO_NULL.getErrorCode(),
 					RegistrationExceptionConstants.REG_BIOMETRIC_DTO_NULL.getErrorMessage());
 
 		ResponseDTO responseDTO = new ResponseDTO();
-		if (validateWithIDA(biometrics, responseDTO)) {
+		if (!validateWithIDA(biometrics, responseDTO)) {
```
Comment (ghost) [inline registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java]: Why is the check negated ?

Comment (ghost) [reply registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java]: resolving it as its removed in the next commit

**Pull request merged**